src/pdl/pdl_interpreter.py

```diff
@@ -1428,18 +1429,36 @@ def generate_client_response_single(
 
 
 def process_call_code(
-    state: InterpreterState, scope: ScopeType, block: CodeBlock, loc: PdlLocationType
-) -> tuple[PdlLazy[Any], LazyMessages, ScopeType, CodeBlock]:
+    state: InterpreterState,
+    scope: ScopeType,
+    block: ArgsBlock | CodeBlock,
+    loc: PdlLocationType,
+) -> tuple[PdlLazy[Any], LazyMessages, ScopeType, ArgsBlock | CodeBlock]:
     background: LazyMessages
-    code_, _, _, block = process_block_of(
-        block,
-        "code",
-        state.with_yield_result(False).with_yield_background(False),
-        scope,
-        loc,
-    )
-    code_s = code_.result()
-    match block.lang:
+    code_a = None
+    code_s = ""
+    lang = ""
+    match block:
+        case ArgsBlock():
+            lang = "command"
+            code_a, _, _, _ = process_block(
```

Comment:
You are loosing the trace form the execution of `block.args`.

Reply:
all fixed, thanks